pkg/snapshot/restorer/restorer.go

```diff
@@ -583,15 +594,24 @@ func getEventsFromDeltaSnapshot(store snapstore.SnapStore, snap snapstore.Snapsh
 }
 
 // getEventsDataFromDeltaSnapshot fetches the events data from delta snapshot from snap store.
-func getEventsDataFromDeltaSnapshot(store snapstore.SnapStore, snap snapstore.Snapshot) ([]byte, error) {
-	rc, err := store.Fetch(snap)
+func (r *Restorer) getEventsDataFromDeltaSnapshot(store snapstore.SnapStore, snap snapstore.Snapshot) ([]byte, error) {
+	reader, err := store.Fetch(snap)
 	if err != nil {
 		return nil, err
 	}
-	defer rc.Close()
+
+	isCompressed := isSnapshotCompressed(snap)
+	if isCompressed {
+		// decompress the snapshots
+		startDeCompressTime := time.Now()
+		reader = compressor.DecompressSnapshots(reader, snap.CompressionSuffix)
+		totalDeCompressTime := time.Now().Sub(startDeCompressTime).Seconds()
+		r.logger.Infof("decompress the snapshot in %v seconds", totalDeCompressTime)
```

Comment:
This logging should happen after `buf.ReadFrom()` call where the stream reading happens -> https://github.com/gardener/etcd-backup-restore/pull/293/files?file-filters%5B%5D=.go#diff-9547d8cc0c155be97c8e3a057b9351b6bf83108543f957fb8bac1adf111ccfaaR614